python/pyspark/streaming/util.py

```diff
@@ -62,9 +63,14 @@ def call(self, milliseconds, jrdds):
             r = self.func(t, *rdds)
             if r:
                 return r._jrdd
-        except Exception:
-            traceback.print_exc()
-            raise
+        except:
+            self.failure = traceback.format_exc()
+
+    def getFailure(self):
+        failure = self.failure
+        # Clear the failure
+        self.failure = None
```

Comment:
Isnt it more intuitive to clear the failure in the beginning of `call`? Otherwise, its weird semantics that successive calls to getFailure will return different things. Also weird that a `get` function has side effects.

Also, better to name it `getLastFailure` to make it more clear that its the failure in the last call.